### PR TITLE
256 promotion sort fix

### PR DESCRIPTION
**Pull request by albertgcsula**
* issue #256: Fix promotion slots drag and drop sorting.

### CODE REVIEW

File: src/plugins/curator/services/PromotionSubscriber.js [hunk 1]
```diff
@@ -195,10 +196,15 @@ export default class PromotionSubscriber extends EventSubscriber {
           const slotNode = slotSchema.createMessage();
           const { name, rendering, widgetRef } = slotData;
 
+          let ref = widgetRef[0];
+          if (typeof ref === 'string') {
```
[the remark this answers is not in the document]
Comment (albertgcsula) [reply src/plugins/curator/services/PromotionSubscriber.js:200]: @arunkarnati I really don't know. I just noticed that this is happening when you drag and drop the field for re-order. When you click the arrow button to sort, it's fine, but when you drag and drop for some reason the noderef turns into a string.

[the remark this answers is not in the document]
Comment (arunkarnati) [reply src/plugins/curator/services/PromotionSubscriber.js:200]: @albertgcsula - I see, perhaps we may need to adjust the _SortableList_ component. For now, please add a comment there, so it is clear.

File: src/plugins/curator/services/PromotionSubscriber.js [hunk 2]
```diff
@@ -195,10 +196,15 @@ export default class PromotionSubscriber extends EventSubscriber {
...
+            ref = NodeRef.fromString(widgetRef[0]);
+          }
+
           slotNode
             .set('name', name)
             .set('rendering', rendering.value ? SlotRendering.create(rendering.value) : null)
-            .set('widget_ref', widgetRef[0]);
+            .set('widget_ref', ref);
```
Comment (arunkarnati) [inline src/plugins/curator/services/PromotionSubscriber.js:207]: @albertgcsula - here's a suggestion:

make the _ref_ a const on Ln 199, and change this line to:

```suggestion
.set('widget_ref', typeof ref === 'string' ? NodeRef.fromString(ref) : ref);
```

Comment (albertgcsula) [reply src/plugins/curator/services/PromotionSubscriber.js]: @arunkarnati okay I made that change for you.

**Review by arunkarnati — APPROVED** [review]
@albertgcsula - LGTM!

**Comment by arunkarnati** [comment]
@albertgcsula - one last thing, either change the base branch to the release-next branch or do the version prep in this branch.

**Comment by albertgcsula** [comment]
> @albertgcsula - one last thing, either change the base branch to the release-next branch or do the version prep in this branch.

@arunkarnati okay i changed the base branch, does that look okay?

**Comment by arunkarnati** [comment]
> > @albertgcsula - one last thing, either change the base branch to the release-next branch or do the version prep in this branch.
> 
> @arunkarnati okay i changed the base branch, does that look okay?

Yes sir! 👍

**Review by gdbrown — APPROVED** [review]

**Pull request merged**